battle_tanks/client/battle_tanks.cpp

```diff
@@ -0,0 +1,421 @@
+                        player->initialize();
+                        player->restore_from_frame(restorer_packet);
+                        game->add_game_object(player);
+                        players.emplace(game_object_id, player);
+                    }
+                }
+                break;
+            }
+            case command_id_server::update_game_frame:
+            {
+                if (player_state == player_game_state::game_session)
+                {
+                    sf::Uint32 game_objects_count;
+                    *packet_ref >> game_objects_count;
+                    for (sf::Uint32 i = 0; i < game_objects_count; ++i)
+                    {
+                        bt::uuid game_object_id;
+                        *packet_ref >> game_object_id;
+
+                        bt::game_object_frame_restorer_packet restorer_packet{ packet_ref };
+                        game->get_game_scene()->get_game_object(game_object_id)->restore_from_frame(restorer_packet);
+                    }
+                }
+                break;
+            }
+            case command_id_server::player_shoot:
+            {
+                if (player_state == player_game_state::game_session)
+                {
+                    bt::uuid player_id;
+                    *packet_ref >> player_id;
+                    bt::uuid bullet_id;
+                    *packet_ref >> bullet_id;
+                    bt::game_object_frame_restorer_packet restorer_packet{ packet_ref };
+                    auto bullet = std::make_shared<bt::bullet_game_object>(
+                        bullet_id,
+                        game->get_physics_body_factory(),
+                        texture_warehouse.load_texture(bt::texture_id::bullet)
+                    );
+                    bullet->initialize();
+                    bullet->restore_from_frame(restorer_packet);
+                    game->add_game_object(bullet);
+                }
+                break;
+            }
+            case command_id_server::delete_game_object:
+            {
+                if (player_state == player_game_state::game_session)
+                {
+                    bt::uuid game_object_id;
+                    *packet_ref >> game_object_id;
+                    game->delete_game_object(game_object_id);
+
+                    if (players.contains(game_object_id))
+                    {
+                        players.erase(game_object_id);
+                    }
+                }
+                break;
+            }
+            }
+        }
+        sf::Event event;
+        while (window->pollEvent(event))
+        {
+            if (event.type == sf::Event::Closed ||
+                (event.type == sf::Event::KeyPressed && event.key.code == sf::Keyboard::Key::Escape))
+            {
+                window->close();
+            }
+
+            if (player_state == player_game_state::main_lobby)
```

Comment:
I believe it's better to move state's handling into its own function/classes

Reply:
done